crates/aptos-node-identity/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{format_err, Result};
+use aptos_types::{chain_id::ChainId, PeerId};
+use once_cell::sync::OnceCell;
+use std::sync::Arc;
+
+/// The global [AptosNodeIdentity]
+static APTOS_NODE_IDENTITY: OnceCell<Arc<AptosNodeIdentity>> = OnceCell::new();
+
+/// Structure that holds information related to a node's identity
+pub struct AptosNodeIdentity {
+    pub chain_id: OnceCell<ChainId>,
+    pub peer_id: Option<PeerId>,
+    pub peer_id_str: Option<String>,
```

Comment:
It feels redundant to store both the `PeerId` and corresponding str. Can we remove this?

Reply:
I am worried to add to logging overhead because a conversion to string needs to happen for each log line.

Reply:
If that's the case, let's just store the string? Do you need to the `PeerId` as well?